### PR TITLE
nrf52: SPI cmddata function mapping wrong for SPI(0,2,3)

### PR DESCRIPTION
**Pull request by btashton**
## Summary
The cmddata function for the SPI Master drivers was mapped to nrf52_spi1cmddata for all drivers not just 1.

## Impact
And build that used SPI masters 0,2,3 would not work correctly with cmddata.

## Testing
Build testing, error is clear.

### CODE REVIEW

**Review by v01d — APPROVED** [review]

**Comment by v01d** [comment]
Lets wait until we fix CI to merge it, if you don't mind.

**Comment by btashton** [comment]
> Lets wait until we fix CI to merge it, if you don't mind.

No worries.  I'm not actually using this logic just noticed it when I was wiring SPI support for the the feather.

**Comment by v01d** [comment]
If you rebase it should pass CI now

**Pull request merged**
